docs/tutorial/transforms.md

```diff
@@ -1,3 +1,193 @@
+@> [1, 2, 3, 4] >> 2 ^ .
+@> ["s", "toe", "thumb"] >> . ++ "nail"
+@> {"A": 10, "B": 42, "C": 100} >> 100 - .
+@> {[-37.8, 145]: 'mel', [-33.9, 151.2]: 'syd', [51.5, -0.1]: 'lcy'} >> //str.upper(.)
+```
+
+#### `>>>`
+
+The `>>>` operator is a variant of the `>>` operator. It performs the same basic
+operation, but also makes the `@` attribute available to the transformation
+expression. This form is special in that it must be a two-argument function.
+
+```arrai
+@> ['red', 'green', 'blue'] >>> \i \c $'${c} = ${i}'
+@> /set cities = {[-37.8, 145]: 'MEL', [-33.9, 151.2]: 'SYD', [51.5, -0.1]: 'LCY'}
+@> cities >>> \[lat, lng] \code $'${code} is at (${lat}, ${lng})'
+```
+
+Note that `>>>` still transforms only the associated value. `@` remains
+unchanged. If you want to transform the entire tuple, use `=>` instead.
+
+### `:>`
+
+The "tuple-arrow" operator, `:>`, requires `lhs` to be a tuple. It applies the
+transform to each attribute value of `lhs` and evaluates to a tuple of the results:
+
+```arrai
+@> (r: 0.5, g: 0.2, b: 0.7) => 1 - .
+```
+
+## Unary forms
+
+The above operator can be used in unary form. For instance, `=> \x 2 + x`. In
+all cases, the `lhs` is implied to be `.`. This property allows the transform
+operators to be chained together to transform deeper structures:
+
+```arrai
+@> [{1, 2}, {2, 3, 4}, {1, 5}] >> => 10 + .
+@> {(r:0.7, g:0, b:0), (r:0.4, g:0.6, b:0), (r:0.5, g:0.5, b:1)} => :> . ^ 2
+```
+
+## Interaction with order and orderby
+
+(The following discussion applies equally to `order` and `orderby`, so we'll
+only discuss `orderby`.)
+
+The `orderby` operator transforms a set into an array:
+
+```arrai
+@> {'red', 'green', 'blue'} orderby .
+@> {(r:0.7, g:0, b:0), (r:0.4, g:0.6, b:0), (r:0.5, g:0.5, b:1)} orderby .r
+@> {(r:0.7, g:0, b:0), (r:0.4, g:0.6, b:0), (r:0.5, g:0.5, b:1)} orderby .g
+```
+
+Things can get a bit confusing when applying `orderby` to a sequence or
+dictionary. It is important to remember that `orderby` always operates on the
+underlying set, never on the abstraction it represents. For instance, consider
+the following expression:
+
+```arrai
+[9, 4, 2, 4] orderby .
+```
+
+At first glance, it seems that this will reorder the numbers of the array.
+However, when you try it out, you'll notice that the output is in fact an array
+of the tuples underlying the original array, not an array of the numbers it
+represents.
+
+In order to rearrange the values accordingly, you must explicitly deal with the
+underlying representation of arrays. While this can be confusing at first, it is
+a result of consistently applying the rule that `orderby` operates on sets. It
+also gives the programmer full control over how ordering should behave. In the
+above case, a question arises: do you want two instances of the number 4 in the
+result? There isn't a single correct answer. If you want an array of all scores,
+you can do it like this:
+
+```arrai
+@> [9, 4, 2, 4] orderby [.@item, .] >> .@item
```

Comment:
maybe we should have a `sort` function for sequenced set?

Reply:
Maybe. I'm not keen to promote extensive use of arrays. I see them as a transitional form that should eventually become of marginal utility.